sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala

```diff
@@ -401,4 +400,14 @@ private[hive] class ClientWrapper(
         client.dropDatabase(db, true, false, true)
       }
   }
+
+  override def shutdown(): Unit = withHiveState {
+    logInfo("Shutting down hive metastore")
+    if (state._1 == Thread.currentThread()) {
```

Comment:
Will this ever be true?

This method is called from the `SparkListener` you're adding, which in a live application is called from the listener bus dispatching thread, which is most probably not the thread that created the `state` object.